src/Infrastructure/UI/Paginator/Factory/OrmQueryFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+declare(strict_types=1);
+
+namespace Landingi\Shared\Infrastructure\UI\Paginator\Factory;
+
+use Doctrine\ORM\Query;
+use Landingi\Shared\Infrastructure\UI\Paginator\Page;
+use Landingi\Shared\Infrastructure\UI\Paginator\Query\QueryLimit;
+use Landingi\Shared\Infrastructure\UI\Paginator\Query\QueryOffset;
+
+class OrmQueryFactory
+{
+    private $query;
+    private $page;
+    private $limit;
+
+    public function __construct(Query $query, Page $page, QueryLimit $limit)
```

Comment:
fabryka jak buduje query to nigdy nie powinna go dostawac w construktorze bo wtedy fabryka jest do uzycia tylko raz. te argument powinny byc dodane do metody build() przy okazji wystarczy samo slowo build wiadomo ze buduje to query bo to QueryFactory ;p a to ze ustawia delimeters bedzie widac po przekazanych argumentach ktore sa wymagane.